docs/spec/distributing.rst

```diff
@@ -10,72 +10,160 @@ Stub files
 
 (Originally specified in :pep:`484`.)
 
-Stub files are files containing type hints that are only for use by
-the type checker, not at runtime.  There are several use cases for
-stub files:
+*Stub files*, also called *type stubs*, provide type information for untyped
+Python packages and modules. Stub files serve multiple purposes:
+
+* They are the only way to add type information to extension modules.
+* They can provide type information for packages that do not wish to
+  add them inline.
+* They can be distributed separately from the implementation.
+  This allows stubs to be developed at a different pace or by different
+  authors, which is especially useful when adding type annotations to
+  existing packages.
+* They can act as documentation, succinctly explaining the external
+  API of a package, without including the implementation or private
+  members.
+
+Stub files use a subset of the constructs used in Python source files, as
+described in :ref:`stub-file-supported-constructs` below. Type checkers should
+parse a stub that uses only such constructs without error and not interpret any
+construct in a contradictory manner. However, type checkers are not required to
+implement checks for all of these constructs and can elect to ignore unsupported
+ones. Additionally, type checkers can support constructs not described here.
+
+Type checkers should have a configurable search path for stub files. If a stub
+file is found, the type checker should not read the corresponding "real" module.
+See :ref:`mro` for more information.
```

Comment:
I guess it predates this PR, but I'm not sure why `mro` is the reference tag for "import resolution ordering".

Reply:
I suppose it stands for "Module Resolution Order" https://peps.python.org/pep-0561/#type-checker-module-resolution-order.